scrapy/extensions/feedexport.py

```diff
@@ -151,7 +151,7 @@ def __init__(self, settings):
         if not self._exporter_supported(self.format):
             raise NotConfigured
         self.store_empty = settings.getbool('FEED_STORE_EMPTY')
-        self.export_fields = settings.getlist('FEED_EXPORT_FIELDS')
+        self.export_fields = settings.getlist('FEED_EXPORT_FIELDS') or None
```

Comment:
> So FEED_EXPORT_FIELDS can't be set to [] by the user with this change, right? 

Yes; `[]` will be interpreted as None.

> Maybe we should fix this in Settings, but I think the only way of knowing if the actual value was set as [] or isn't set at all is by checking with settings.get() before.

Yes, maybe it is better to fix it in settings, but fixing it in settings is more controversial and requires more discussion because it affects more code. `getlist` always returning a list kind of make sense, but on the other hand preserving `default` as-is also kind of make sense. We may use another sentinel instead of None for default `default` value to keep None - but does it worth the troubles, what are real use cases? That's what I mean by 'requires more discussion'. 

This issue is a serious regression, so I think we should fix it without fixing issues with Settings if fixing Settings take time.